webapp/src/components/Modal/AuctionModal/AuctionModal.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Loader, Button } from 'semantic-ui-react'
+
+import ContractLink from 'components/ContractLink'
+import { t, T } from '@dapps/modules/translation/utils'
+import { dismissAuctionModal, isAuthorized } from 'modules/auction/utils'
+import { authorizationType } from 'components/types'
+
+import BaseModal from '../BaseModal'
+
+import './AuctionModal.css'
+
+export default class AuctionModal extends React.PureComponent {
+  static propTypes = {
+    ...BaseModal.propTypes,
+    authorization: authorizationType,
+    onAuthorize: PropTypes.func.isRequired
+  }
+
+  handleOnAuthorize = () => {
+    const { onClose, onAuthorize } = this.props
+    onClose()
+    onAuthorize()
```

Comment:
Is it ok to close and authorize? I'm wondering if changing the order has an effect

Reply:
it should be the same, the onClose only changes a boolean in the global redux state. I wanted to close it first because `onAuthorize` is a redirect. So you close and _then_ see something else.
In practice it's so fast that it's the same thing